File: benchmarks/python/03_patient_profiles.py
"""Benchmark 03: Patient Profiles — omopy.profiles.add_demographics()"""
import sys; sys.path.insert(0, "benchmarks/python")
from helpers import connect_cdm, save_result, save_timing, Timer

print("=== 03: Patient Profiles ===")
t = Timer()
cdm = connect_cdm()

from omopy.generics import Codelist
from omopy.connector import generate_concept_cohort_set
from omopy.profiles import add_demographics

cdm = generate_concept_cohort_set(cdm, Codelist({"coronary_artery": [317576]}), name="target_cohort")
cohort = cdm["target_cohort"]

enriched = add_demographics(cohort, cdm)
df = enriched.collect()
save_result(df.head(100), "03_patient_profiles")
save_timing("03_patient_profiles", t.elapsed())
print(f"Done in {t.elapsed():.2f} seconds")

File: benchmarks/r/03_patient_profiles.R
# Benchmark 03: Patient Profiles
# R equivalent: PatientProfiles::addDemographics()

source("benchmarks/r/00_helpers.R")
library(PatientProfiles)
cat("=== 03: Patient Profiles ===\n")
t0 <- proc.time()

con <- dbConnect(duckdb(), dbdir = DB_PATH)
cdm <- cdmFromCon(con = con, cdmSchema = "main", writeSchema = "main", cdmName = "synthea_1k")

cdm <- generateConceptCohortSet(
  cdm = cdm,
  conceptSet = list(coronary_artery = 317576),
  name = "target_cohort"
)

enriched <- cdm$target_cohort |>
  addDemographics() |>
  collect()

# Save first 100 rows
save_result(head(enriched, 100), "03_patient_profiles")
elapsed <- (proc.time() - t0)[["elapsed"]]
save_timing("03_patient_profiles", elapsed)
cdmDisconnect(cdm)
cat("Done in", elapsed, "seconds\n")